### PR TITLE
Added ui for contact list

### PR DESCRIPTION
**Pull request by ViachaslauPratasenia**

### CODE REVIEW

File: app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        } else {
+            viewHolder.contactButton.setBackgroundResource(R.drawable.follow);
+        }
+        viewHolder.contactButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                item = list.get(i);
+                if(item.isFriend()){
+                    item.setFriend(false);
+                    viewHolder.contactButton.setBackgroundResource(R.drawable.follow);
```
Comment (YodaMike) [inline app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java]: @ViachaslauPratasenia what if there would be an update failure?

File: app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import java.util.ArrayList;
+
+import de.hdodenhof.circleimageview.CircleImageView;
+
+public class ContactListAdapter extends RecyclerView.Adapter<ContactListAdapter.ViewHolder> {
+
+    private ArrayList<Contact> list = new ArrayList<>();
+    private Context context;
+    //TODO init manager
+    private ContactManager contactManager;
```
Comment (YodaMike) [inline app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java]: @ViachaslauPratasenia I didn't found a place where you assign a value to the `contactManager` field

File: app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            viewHolder.contactButton.setBackgroundResource(R.drawable.follow);
+        }
+        viewHolder.contactButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                item = list.get(i);
+                if(item.isFriend()){
+                    item.setFriend(false);
+                    viewHolder.contactButton.setBackgroundResource(R.drawable.follow);
+                    contactManager.update(item, onSuccess, onFailure);
```
Comment (YodaMike) [inline app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java]: @ViachaslauPratasenia could be nullpointer exception

File: app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactsFragment.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+import com.heapixLearn.discovery.R;
+import java.util.ArrayList;
+
+public class ContactsFragment extends Fragment implements SearchView.OnQueryTextListener {
+
+    View contactsFragment;
+    private ArrayList<Contact> contacts = new ArrayList<>();
+    //TODO init manager
+    private ContactManager contactManager;
```
Comment (YodaMike) [inline app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactsFragment.java]: @ViachaslauPratasenia you don't assign a value to this field

File: app/src/main/java/com/heapixLearn/discovery/ContactManager.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.heapixLearn.discovery;
+
+import com.heapixLearn.discovery.ui.contact.Contact;
+
+import java.util.ArrayList;
+
+public class ContactManager implements com.heapixLearn.discovery.ui.contact.ContactManager {
```
Comment (Agent00krosh) [inline app/src/main/java/com/heapixLearn/discovery/ContactManager.java]: replace to logic directory

File: app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java [hunk 6]
```diff
@@ -87,20 +93,45 @@ public void onBindViewHolder(@NonNull final ViewHolder viewHolder, final int i)
...
             @Override
             public void onClick(View v) {
-                item = list.get(i);
-                if(item.isFriend()){
-                    item.setFriend(false);
-                    viewHolder.contactButton.setBackgroundResource(R.drawable.follow);
-                    contactManager.update(item, onSuccess, onFailure);
+                if(contactManager != null){
+                    item = list.get(i);
+                    if(item.isFriend()){
```
Comment (Agent00krosh) [inline app/src/main/java/com/heapixLearn/discovery/ui/contact/ContactListAdapter.java]: why do you need if?

**Pull request merged**